### PR TITLE
Target p2panda main

### PR DESCRIPTION
**Pull request by sandreae**
## 📋 Checklist

- [x] Add tests that cover your changes
- [x] Add this PR to the _Unreleased_ section in `CHANGELOG.md`
- [x] Link this PR to any issues it closes
- [x] New files contain a SPDX license header

### CODE REVIEW

**Comment by sandreae** [comment]
Ah ha, yes, now that we aren't sorting the content of `RelationList` and `PinnedRelationList` the problem is back. I think I understand it now though, maybe this is what you though @adzialocha. Insert list items into the db individually, and so when they are retrieved and reconstructed as a list, order is not necessarily preserved. Hmmm... need to think on the best way to handle this, guess we need to index list items.

**Comment by adzialocha** [comment]
> Ah ha, yes, now that we aren't sorting the content of `RelationList` and `PinnedRelationList` the problem is back. I think I understand it now though, maybe this is what you though @adzialocha. We insert list items into the db as individual rows, and so when they are retrieved and reconstructed as a list, order is not necessarily preserved. Hmmm... need to think on the best way to handle this, guess we need to index list items.

Yeah, I was thinking the same .. maybe there is a crate for that but so far I couldn't find anything: https://www.reddit.com/r/rust/comments/cz6te1/async_preventing_reordering_of_streams_and_futures/

**Comment by sandreae** [comment]
I implemented indexing the operation rows on insertion and then ordering them by index in retrieval. See what you think, I don't know if that's a hacky fix 😅

**Comment by sandreae** [comment]
I'm gunna merge this for now, we can apply an alternative solution to the field ordering if anyone's not happy with this :pray:

File: aquadoggo/src/db/utils.rs [hunk 1]
```diff
@@ -622,10 +622,10 @@ mod tests {
                 .get("many_profile_pictures")
                 .unwrap(),
             &OperationValue::RelationList(RelationList::new(vec![
-                "0020bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb"
+                "0020aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
```
Comment (cafca) [inline aquadoggo/src/db/utils.rs:625]: aaaaaaaaaaaaa 🙀🙀🙀🙀

File: aquadoggo/src/db/stores/document.rs [hunk 2]
```diff
@@ -220,13 +224,17 @@ impl DocumentStore for SqlStorage {
...
+            ORDER BY
+                operation_fields_v1.list_index ASC
             ",
         )
         .bind(id.as_str())
         .fetch_all(&self.pool)
         .await
         .map_err(|e| DocumentStorageError::FatalStorageError(e.to_string()))?;
 
+        println!("{:#?}", document_view_field_rows);
```
[the remark this answers is not in the document]
Comment (sandreae) [reply aquadoggo/src/db/stores/document.rs:236]: Ah! Thanks!!

**Pull request merged**